Events/InteractionEvents.cs

```diff
@@ -200,6 +200,152 @@ await LogChannelHelper.LogDeletedMessagesAsync(
+                await e.Interaction.DeferAsync(ephemeral: true);
+                
+                // Get member
+                var member = await e.Guild.GetMemberAsync(e.User.Id);
+                
+                // Map role select options to role IDs
+                var insiderRoles = new Dictionary<string, ulong>
+                {
+                    { "insiders-info-w11-canary", cfgjson.UserRoles.InsiderCanary },
+                    { "insiders-info-w11-dev", cfgjson.UserRoles.InsiderDev },
+                    { "insiders-info-w11-beta", cfgjson.UserRoles.InsiderBeta },
+                    { "insiders-info-w11-rp", cfgjson.UserRoles.InsiderRP },
+                    { "insiders-info-w10-rp", cfgjson.UserRoles.Insider10RP },
+                    { "insiders-info-pt", cfgjson.UserRoles.PatchTuesday }
+                };
+                
+                // Get a list of the member's current roles that we can add to or remove from
+                // Then we can apply this in a single request with member.ReplaceRolesAsync to avoid making repeated member update requests
+                List<DiscordRole> memberRoles = member.Roles.ToList();
+                
+                var selection = e.Values.Select(x => insiderRoles[x]).ToList();
+                
+                foreach (var roleId in insiderRoles.Values)
+                {
+                    var role = await e.Guild.GetRoleAsync(roleId);
+                    
+                    if (selection.Contains(roleId))
+                    {
+                        // Member should have the role
+                        if (!memberRoles.Contains(role))
+                            memberRoles.Add(role);
+                    }
+                    else
+                    {
+                        // Member should not have the role
+                        if (memberRoles.Contains(role))
+                            memberRoles.Remove(role);
+                    }
+                }
+                
+                await member.ReplaceRolesAsync(memberRoles, "Applying Insider roles chosen in #insiders-info");
+                
+                await e.Interaction.CreateFollowupMessageAsync(new DiscordFollowupMessageBuilder().WithContent($"{cfgjson.Emoji.Success} Your Insider roles have been updated!").AsEphemeral(true));
+            }
+            else if (e.Id == "insiders-info-chat-btn-callback")
+            {
+                // Button in #insiders-info that checks whether user has 'insiderChat' role and asks them to confirm granting/revoking it
+                
+                // Defer
+                await e.Interaction.DeferAsync(ephemeral: true);
+                
+                // Get member
+                var member = await e.Guild.GetMemberAsync(e.User.Id);
+                
+                var insiderChatRole = await e.Guild.GetRoleAsync(cfgjson.UserRoles.InsiderChat);
+                if (member.Roles.Contains(insiderChatRole))
```

Comment:
This should also check if you have any of the insider update roles and mention that you already have channel access